src/backend/gl/src/info.rs

```diff
@@ -233,68 +225,88 @@ impl Info {
     pub fn is_version_or_extension_supported(&self, major: u32, minor: u32, ext: &'static str) -> bool {
         self.is_version_supported(major, minor) || self.is_extension_supported(ext)
     }
+
+    pub fn is_any_extension_supported(&self, exts: &[&'static str]) -> bool {
+        exts.iter().any(|e| self.extensions.contains(e))
+    }
+
+    /// Returns `true` if current version is greater than required version for profile
+    /// or if extension string is present in extension list
+    pub fn is_supported(&self, core: Option<(u32, u32)>, es: Option<(u32, u32)>, exts: &[&'static str]) -> bool {
+        if self.version.is_embedded { es } else { core }
+          .map_or(false, |v| self.version >= Version::new(v.0, v.1, None, "")) ||
+          self.is_any_extension_supported(exts)
+    }
+}
+
+macro_rules! cap_support {
+    ( $info:expr => core $version:expr ) => { $info.is_version_supported($version.0, $version.1) };
+    ( $info:expr => es $version:expr ) => { $info.is_embedded_version_supported($version.0, $version.1) };
+    ( $info:expr => [ $( $extensions:expr ),* ] ) => { $info.is_any_extension_supported(&[$($extensions),*]) };
+
+    ( $info:expr $(, core $vc:expr)* $(, es $ve:expr)* $(, [ $($ext:expr),* ])* ) => {
+        $( cap_support!($info => core $vc) | )*
+        $( cap_support!($info => es $ve) | )*
+        $( cap_support!($info => [ $($ext),* ] ) | )*
+        false
+    };
 }
 
 /// Load the information pertaining to the driver and the corresponding device
 /// capabilities.
 pub fn get(gl: &gl::Gl) -> (Info, Capabilities, PrivateCaps) {
     let info = Info::get(gl);
-    let tessellation_supported =           info.is_version_or_extension_supported(4, 0, "GL_ARB_tessellation_shader");
+    let tessellation_supported =           cap_support!(info, core (4,0),
```

Comment:
this is much better! I think we can improve here by not using a macro. Instead, you could have
```
enum RequiredVersion {
    Core(u32,u32),
    Es(u32, u32),
}
```
which is included as `use RequiredVersion::*`, so that the ergonomics of using it inside this function stays as good as it is with the macro